services/src/main/java/org/keycloak/protocol/oid4vc/issuance/OID4VCIssuerWellKnownProvider.java

```diff
@@ -189,19 +185,34 @@ private static List<String> getSupportedEncryptionMethods() {
 
     /**
      * Returns the supported compression methods from realm attributes.
+     *
+     * Note: Keycloak's JWE implementation currently only has placeholder support for compression
+     * in the JWEHeader class, but no actual compression/decompression logic is implemented.
+     * The compression algorithm field exists but is not processed during JWE encoding/decoding.
+     *
+     * TODO: Implement JWE compression support when Keycloak core adds compression functionality
      */
     private static List<String> getSupportedCompressionMethods() {
-        // For now, return empty list as compression is optional
+        // Keycloak JWE implementation lacks compression support - only header placeholder exists
         return List.of();
     }
 
     /**
      * Returns the encryption JWKS from realm keys.
+     * Filters the realm JWKS to include only encryption keys.
      */
     private static List<JWK> getEncryptionJwks(KeycloakSession session) {
-        // TODO: Implement proper JWK conversion from KeyWrapper
-        // For now, return empty list as this is optional
-        return List.of();
+        RealmModel realm = session.getContext().getRealm();
+        JSONWebKeySet realmJwks = JWKSServerUtils.getRealmJwks(session, realm);
+
+        if (realmJwks.getKeys() == null) {
+            return List.of();
+        }
+
+        return List.of(realmJwks.getKeys())
+                .stream()
+                .filter(jwk -> jwk.getPublicKeyUse() != null && "enc".equals(jwk.getPublicKeyUse()))
+                .collect(Collectors.toList());
```

Comment:
This could be simplified.

```suggestion
        return Stream.of(realmJwks.getKeys())
                .filter(jwk -> KeyUse.ENC.getSpecName().equals(jwk.getPublicKeyUse()))
                .toList();
```